### ISSUE TITLE
[C# 12]: New Experimental Feature - Interceptors 

### ISSUE THREAD

**Issue by BillWagner** [dotnet/docs#35878]
### Visual Studio release

Other (please put exact version in description textbox)

### Link to speclet

- [Speclet](https://github.com/dotnet/roslyn/blob/main/docs/features/interceptors.md)

### Link to Pull request

- [PR](https://github.com/dotnet/roslyn/pull/68564)

### New behavior

Added in 17.7, preview 3.

Because this is an experimental feature, at this time, we're only adding a note in the What's new page, and covering new compiler warnings and errors.

An *interceptor* is a method which can declaratively substitute a call to an *interceptable* method with a call to itself at compile time. This substitution occurs by having the interceptor declare the source locations of the calls that it intercepts. This provides a limited facility to change the semantics of existing code by adding new code to a compilation (e.g. in a source generator).

### Areas affected

- [X] **Add to What's new in C# 12**: Add links to updated docs in the What's new in C# 12 article.
- [ ] **Update language reference**: Detail the areas in the language reference that will be updated for this feature.
- [X] **Update compiler messages**: Add new (or updated) pages for new compiler messages. Include related message.
- [ ] *Include in *Everyday C#* section**: Is this new feature included in Everyday C#?
- [ ] **Tutorials**: Are new tutorials required for this feature? If so, detail below

### Language reference tasks

- N/A for an experimental feature. We'll add more as the feature spec coalesces more.

---
[Associated WorkItem - 117224](https://dev.azure.com/msft-skilling/Content/_workitems/edit/117224)

**Comment by jcouv** [comment]
nit: this is not technically a language feature, it is not tied to C# 12. That said, it's probably fine to bundle in the "what's new in C# 12" post. I'm not sure how we handled such features in the past.

**Comment by BillWagner** [comment]
> I'm not sure how we handled such features in the past.

Bundling this with C# 12 is similar to what I did for static abstract methods in interfaces: I added them to "what's new in C#10" as an experimental feature. Then, when the design was finalized for C# 11, I moved that feature to the "what's new in C# 11" instead. That way, customers could find it sooner, but were notified that it was an experimental feature until C# 11.
